src/net_processing.cpp

```diff
@@ -3273,17 +3273,14 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
         if (greatest_common_version >= WTXID_RELAY_VERSION && m_txreconciliation) {
```

Comment:
Not sure if that is better, just an idea, feel free to ignore:

```diff
diff --git i/src/net_processing.cpp w/src/net_processing.cpp
index 376a263183..0ed7781b2c 100644
--- i/src/net_processing.cpp
+++ w/src/net_processing.cpp
@@ -3265,26 +3265,27 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             (fRelay || (peer->m_our_services & NODE_BLOOM))) {
             auto* const tx_relay = peer->SetTxRelay();
             {
                 LOCK(tx_relay->m_bloom_filter_mutex);
                 tx_relay->m_relay_txs = fRelay; // set to true after we get the first filter* message
             }
-            if (fRelay) pfrom.m_relays_txs = true;
-        }
-
-        if (greatest_common_version >= WTXID_RELAY_VERSION && m_txreconciliation) {
-            // Per BIP-330, we announce txreconciliation support if:
-            // - protocol version per the peer's VERSION message supports WTXID_RELAY;
-            // - transaction relay is supported per the peer's VERSION message (see m_relays_txs);
-            // - this is not a block-relay-only connection and not a feeler (see m_relays_txs);
-            // - this is not an addr fetch connection;
-            // - we are not in -blocksonly mode.
-            if (pfrom.m_relays_txs && !pfrom.IsAddrFetchConn() && !m_ignore_incoming_txs) {
-                const uint64_t recon_salt = m_txreconciliation->PreRegisterPeer(pfrom.GetId());
-                m_connman.PushMessage(&pfrom, msg_maker.Make(NetMsgType::SENDTXRCNCL,
-                                                             TXRECONCILIATION_VERSION, recon_salt));
+            if (fRelay) {
+                pfrom.m_relays_txs = true;
+
+                if (greatest_common_version >= WTXID_RELAY_VERSION &&
+                    m_txreconciliation &&
+                    !pfrom.IsAddrFetchConn() && !m_ignore_incoming_txs) {
+                    // Per BIP-330, we announce txreconciliation support if:
+                    // - protocol version per the peer's VERSION message supports WTXID_RELAY;
+                    // - transaction relay is supported per the peer's VERSION message (see m_relays_txs);
+                    // - this is not a block-relay-only connection and not a feeler (see m_relays_txs);
+                    // - this is not an addr fetch connection;
+                    // - we are not in -blocksonly mode.
+                    const uint64_t recon_salt = m_txreconciliation->PreRegisterPeer(pfrom.GetId());
+                    m_connman.PushMessage(&pfrom, msg_maker.Make(NetMsgType::SENDTXRCNCL, TXRECONCILIATION_VERSION, recon_salt));
+                }
             }
         }
 
         m_connman.PushMessage(&pfrom, msg_maker.Make(NetMsgType::VERACK));
 
         // Potentially mark this peer as a preferred download peer.
```

Reply:
Not sure about this — it's kinda difficult either way :) Will see what others say.